src/signals/IncidentMap/components/FilterPanel/FilterCategories.tsx

```diff
@@ -0,0 +1,92 @@
+/* SPDX-License-Identifier: MPL-2.0 */
+/* Copyright (C) 2022 Gemeente Amsterdam */
+import { useState } from 'react'
+
+import { ChevronDown } from '@amsterdam/asc-assets'
+
+import type { Filter } from '../../types'
+import { FilterCategory } from './FilterCategory'
+import {
+  SectionWrapper,
+  WrapperFilterCategoryWithIcon,
+  InvisibleButton,
+  SubSection,
+} from './styled'
+
+export interface Props {
+  onToggleCategory: (option: Filter[], select: boolean) => void
+  filter: Filter
+}
+
+export const FilterCategories = ({ onToggleCategory, filter }: Props) => {
+  const [showSubsection, setShowSubsection] = useState<boolean>(true)
+  const { name, filterActive, _display, icon, subCategories } = filter
+
+  const showAfvalAndWegenCategories = (name: string) => {
+    return name === 'Afval' || name === 'Wegen, verkeer, straatmeubilair'
+  }
+
+  return (
+    <>
+      {showAfvalAndWegenCategories(name) && subCategories ? (
+        <SectionWrapper key={`section-${name}`}>
+          <WrapperFilterCategoryWithIcon>
+            <FilterCategory
+              onToggleCategory={(checked: boolean) => {
+                onToggleCategory([filter].concat(subCategories), checked)
```

Comment:
You have four different implementations of "onToggleCategory" now. I think it is better to put all that logic into the onToggle function it self so it can handle all three different scenarios.